### PR TITLE
Make IOS.setTaskScheduler() as public

### PR DESCRIPTION
**Pull request by artembilan**
* Make `IntegrationObjectSupport.setTaskScheduler()` as `public` and
remove all the overrides for visibility
* Fix Sonar smells for all the affected classes

### CODE REVIEW

File: spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractEndpoint.java [hunk 1]
```diff
@@ -125,7 +119,7 @@ protected void onInit() {
 	}
 
 	@Override
-	public void destroy() throws Exception {
+	public void destroy() {
```
Comment (garyrussell) [inline spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractEndpoint.java]: We can't do this - breaking change for anyone that has subclassed and implemented `destroy()`.

I suggest we use `// NOSONAR TODO: remove throws in 5.2`, which removing it from our subclasses.

Comment (artembilan) [reply spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractEndpoint.java]: OK. 
Do we remove from our subclasses now as I did in this PR?

Sorry, if I misunderstood you...

**Comment by artembilan** [comment]
Is the latest commit OK?

**Comment by garyrussell** [comment]
LGTM; let's see if Mr. Travis agrees.

**Comment by artembilan** [comment]
OK. It's green finally! 😄

**Pull request merged**